thunder/core/symbol.py

```diff
@@ -525,8 +525,11 @@ def __eq__(self, other):
 
         return (self.sym, self._var_args, self._var_output) == (other.sym, other._var_args, other._var_output)
 
-    def rhs(self):
-        return BoundSymbolRHS(self)
+    @functools.cached_property
```

Comment:
This should be just `@property`. BoundSymbol itself is not immutable and having cached_property is unsafe. Take a look at this simplified example:
```py
from dataclasses import dataclass
import functools

@dataclass
class Test:
    test_attr: int

    @functools.cached_property
    def test(self):
        return self.test_attr

test = Test(2)
print(test.test)
test.test_attr = 3.0
print(test.test)
assert not test.test == 2
```

Reply:
Nevermind, we have other similar usages already with https://github.com/Lightning-AI/lightning-thunder/blob/8309fc015c1ea9d684337cdcf95138524cd8645c/thunder/core/symbol.py#L436-L437

BoundSymbol should be just made a frozen dataclass.